mm2src/mm2_main/tests/mm2_tests/mm2_tests_inner.rs

```diff
@@ -1674,9 +1674,10 @@ fn test_cancel_all_orders() {
     ));
     assert!(!order_path.exists());
 
-    let pause = 3;
-    log!("Waiting ({} seconds) for Bob to cancel the order…", pause);
-    thread::sleep(Duration::from_secs(pause));
+    block_on(mm_alice.wait_for_log(5., |log| {
+        log.contains(&format!("Maker order {} was recently cancelled, ignoring", uuid))
+    }))
+    .unwrap();
```

Comment:
electrum PR fails here. passes in `test_cancel_order` though.

Reply:
Try the old test without this log, it will work for sure. Sometimes the messages arrive in the right order it seems.

Reply:
Will fix this to not rely on the log in case messages arrive in the right order (which happens sometimes of course)

Reply:
Something interesting happened @mariocynicys , `test_cancel_all_orders` failed here after I cherry picked the changes to expirable map from your branch https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/11100978215/job/30837774256?pr=2232#step:7:1547 . I will solve it by this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2232#discussion_r1780529737 but we found the changes that makes it fail there (not due to old out of order messages reason but I mean this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2232#discussion_r1779464754)

Reply:
> idk i think macos used to fail it sometimes in the past, but can't verify since logs get purged by github. this failure isn't consistent though since it only happened on macos.
and the test still fails after reverting the expirable map changes :(

No problem, will fix it by not relying on the log and everything should be fine.

Reply:
This https://github.com/KomodoPlatform/komodo-defi-framework/pull/2232/commits/87c53f81ed7cc727f6af4715a6289bd82d5a8bc1 should work on the electrums PR, the test also covers the case of out of order messages when it happens, not in every run though :)

Reply:
macOS still unstable and fails sometimes. I think it's another issue and the test is fine.